apps/wistia-videos/src/components/Field.tsx

```diff
@@ -31,37 +33,63 @@ const StyledImageContainer = styled.div`
 
 const Field = (props: FieldProps) => {
   const { sdk } = props;
-  const [data, updateData] = useState<WistiaItem[] | []>([]);
+  const [data, updateData] = useState<WistiaVideo[] | []>([]);
   const [selectedIds, setIds] = useState<number[]>([]);
-  const [dropdownData, filterDropdownData] = useState<WistiaItem[] | []>([]);
+  const [dropdownData, filterDropdownData] = useState<WistiaVideo[] | []>([]);
   const [loading, updateLoadingStatus] = useState(true);
+  const [end, setEnd] = useState(100);
   const [error, setError] = useState(null || '');
   sdk.window.startAutoResizer();
 
+  const fetchPagedVideos = async (
+    projectIds: string[],
+    parameters: any,
+    page?: number
+  ): Promise<WistiaVideo[]> => {
+    return await fetchVideos(projectIds, parameters.apiBearerToken, page);
+  };
+
   // Set inital state based on field values
   useEffect(() => {
+    const fetchInitialVideos = async () => {
+      try {
+        const parameters: any = sdk.parameters.installation;
+        const projectIds = await getProjectIds(
+          parameters.excludedProjects,
+          parameters.apiBearerToken
+        );
+        let videos = await fetchPagedVideos(projectIds, parameters);
+        if (videos.length === 100) {
```

Comment:
does their API not return a `total` or something – this seems very brute forced :(

Reply:
It does not! Patrick showed me yesterday your only options are a starting value, and a page count, which can be 100 at max. You also don't get any metadata around the total number of records in your request

Reply:
I took another look at this, and I think I found a workaround we can use that's less brute forced 👍